File: swift/1656_DesignAnOrderedStream.swift

class OrderedStream {
    var stream: [String]
    var idx = 1
    init(_ n: Int) {
        stream = [String](repeating: "", count: 1001)
    }
    
    func insert(_ id: Int, _ value: String) -> [String] {
        stream[id] = value
        var res = [String]()
        var i = idx
        while stream[i] != "" {
            res.append(stream[i])
            i += 1
        }
        idx = i
        return res
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * let obj = OrderedStream(n)
 * let ret_1: [String] = obj.insert(id, value)
 */

File: python/1656_DesignAnOrderedStream.py
class OrderedStream:

    def __init__(self, n: int):
        self.stream = [""] * 1001
        self.head = 1

    def insert(self, idx: int, value: str) -> List[str]:
        self.stream[idx] = value
        res = []
        i = self.head
        while self.stream[i] != "":
            res.append(self.stream[i])
            i += 1
        self.head = i
        return res
